tests/sentry/api/test_base.py

```diff
@@ -55,6 +55,29 @@ def test_basic_cors(self):
 
         assert response["Access-Control-Allow-Origin"] == "http://example.com"
 
+    def test_invalid_cors_without_auth(self):
+        request = HttpRequest()
+        request.method = "GET"
+        request.META["HTTP_ORIGIN"] = "http://example.com"
+
+        with self.settings(SENTRY_ALLOW_ORIGIN="https://sentry.io"):
```

Comment:
I wasn't able to find this config option in our production configuration. It looks like we're inheriting the default of None defined in `src/sentry/conf/server.py`. It looks like that will result in unauthenticated requests always getting 400 responses. Don't signups require unauthenticated requests?

Reply:
without it configured i believe it maps to `*` though.. why would we not have this configured... i will look more thoroughly as maybe i mis-remember what that setting is implying.

its also unrelated to authenticatd/unauthenticated, it applies globally to any request which isnt using an api token, which primarily is session-based requests

Reply:
actually the code comment suggests this might be focused on /api/store. will sync w/ relay team to understand how this works these days.. might just need a new setting (or to hijack another setting that gives us domains).

Reply:
There's also this setting https://github.com/getsentry/getsentry/blob/560eddcde3627c4faa7287bb1349c38b458d3d17/getsentry/conf/settings/prod.py#L198

Reply:
Though that might be pretty unrelated to CORS

Reply:
Repurposing the setting should be ok. Do we have any unauthenticated endpoints that are used in self-hosted scenarios that could start to fail in environments that don't have this configured? Things like password reset, accepting invitations come to mind as possibly impacted flows.

Reply:
Only applies to JS requests so there shouldn't be a concern. Those pages are still server rendered AFAIK. Mostly I doubt people are actually setting this in their self-hosted installs. We could rename it, just didn't see a lot of value.